src/http/representation/RepresentationMetadata.ts

```diff
@@ -304,6 +306,52 @@ export class RepresentationMetadata {
     return this;
   }
 
+  private setContentTypeParams(input: ContentType | string | undefined): void {
+    if (input === undefined) {
+      this.removeContentTypeParameters();
+      return;
+    }
+    if (typeof input === 'string') {
+      input = parseContentTypeWithParameters(input);
+    }
+
+    // Make sure complete Content-Type RDF structure is gone
+    this.removeContentTypeParameters();
+
+    Object.entries(input.parameters ?? []).forEach(([ paramKey, paramValue ], idx): void => {
+      const paramNode = DataFactory.blankNode(`parameter${idx + 1}`);
+      this.addQuad(this.id, SOLID_META.terms.ContentTypeParameter, paramNode);
+      this.addQuad(paramNode, RDFS.terms.label, paramKey);
+      this.addQuad(paramNode, SOLID_META.terms.value, paramValue);
+    });
+  }
+
+  /**
+   * Parse the internal RDF structure to retrieve the Record with ContentType Parameters.
+   * @returns An Record&lt;string,string&gt; object with Content-Type parameters and their values.
+   */
+  private getContentTypeParams(): Record<string, string> | undefined {
```

Comment:
```suggestion
  private getContentTypeParams(): Record<string, string> {
```
I think returning an empty object instead of undefined makes more sense if there are no results. This makes it easier for classes calling `contentTypeObject` as they don't first always have to check if the object is present before they check if a specific key is there. This also allows removing the check for empty param lists.